tests/IntegrationTests/WorkflowExecutor.IntegrationTests/StepDefinitions/WorkflowTaskArtifactStepDefinitions.cs

```
@@ -0,0 +1,90 @@
+            }
+            else
+            {
+                PayloadId = DataHelper.WorkflowInstances[0].PayloadId;
+            }
+
+            _outputHelper.WriteLine($"Retrieving workflow instance using the payloadid={PayloadId}");
+            var workflowInstances = DataHelper.GetWorkflowInstances(1, PayloadId);
+            _outputHelper.WriteLine("Retrieved workflow instance");
+
+            if (workflowInstances != null)
+            {
+                foreach (var workflowInstance in workflowInstances)
+                {
+                    var workflowRevision = DataHelper.WorkflowRevisions.OrderByDescending(x => x.Revision).FirstOrDefault(x => x.WorkflowId.Equals(workflowInstance.WorkflowId));
+
+                    if (workflowRevision != null)
+                    {
+                        foreach (var task in workflowInstance.Tasks)
+                        {
+                            var seededTask = workflowInstance.Tasks.FirstOrDefault(x => x.ExecutionId.Equals(task.ExecutionId));
+
+                            if (seededTask == null)
+                            {
+                                var workflowTask = workflowRevision.Workflow.Tasks.FirstOrDefault(x => x.Id.Equals(task.TaskId));
+
+                                if (workflowTask != null)
+                                {
+                                    Assertions.AssertInputArtifacts(workflowTask, PayloadId, task);
+                                }
+                                else
+                                {
+                                    throw new Exception($"Workflow Revision Task or {task.TaskId} not found!");
+                                }
+                            }
+                        }
+                    }
+                    else
+                    {
+                        throw new Exception($"Workflow not found for workflowId {workflowInstance.WorkflowId}");
```

If you just do .First(...) rather than FirstOrDefault it will throw when they are null seems like you are throwing if its null so might be better as a First and wont need to do throws after?

---

Thanks Lillie, I'll change that